index.d.ts

```diff
@@ -626,32 +626,62 @@ export interface Style {
 
 /**
  * Interface for defining unsupported properties in the parsers.
- * Value of each property must either be 'undefined', or an object whose (sub-)subproperties have the value 'undefined'.
  */
 export interface UnsupportedProperties {
-  ScaleDenominator?: 'unsupported';
-  Filter?: {
-    '&&'?: 'unsupported';
-    '!'?: 'unsupported';
-    '||'?: 'unsupported';
-    '>'?: 'unsupported';
-    '>='?: 'unsupported';
-    '=='?: 'unsupported';
-    '<'?: 'unsupported';
-    '<='?: 'unsupported';
-    '!='?: 'unsupported';
-    '*='?: 'unsupported';
+  ScaleDenominator?: SupportDef;
+  Filter?: SupportDef | {
+    '&&'?: SupportDef;
+    '!'?: SupportDef;
+    '||'?: SupportDef;
+    '>'?: SupportDef;
+    '>='?: SupportDef;
+    '=='?: SupportDef;
+    '<'?: SupportDef;
+    '<='?: SupportDef;
+    '!='?: SupportDef;
+    '*='?: SupportDef;
   };
-  Symbolizer?: {
-    PointSymbolizer?: any;
-    LineSymbolizer?: any;
-    FillSymbolizer?: any;
-    MarkSymbolizer?: any;
-    IconSymbolizer?: any;
-    RasterSymbolizer?: any;
+  Symbolizer?: SupportDef | {
+    PointSymbolizer?: {
+      [key in keyof Required<PointSymbolizer>]?: SupportDef
+    };
+    LineSymbolizer?: {
+      [key in keyof Required<LineSymbolizer>]?: SupportDef
+    };
+    FillSymbolizer?: {
+      [key in keyof Required<FillSymbolizer>]?: SupportDef
+    };
+    MarkSymbolizer?: {
+      [key in keyof Required<MarkSymbolizer>]?: SupportDef
+    };
+    IconSymbolizer?: {
+      [key in keyof Required<IconSymbolizer>]?: SupportDef
+    };
+    RasterSymbolizer?: {
+      [key in keyof Required<RasterSymbolizer>]?: SupportDef
+    };
   };
 }
 
+/**
+ * Level of support for a functionality.
+ */
+export type SupportLevel = 'full' | 'partial' | 'none';
```

Comment:
Does it make sense to have a `full` support property for UNsupported properties?

Reply:
I was not sure about that. I'll remove it.